src/utils/check-page-for-error.js

```diff
@@ -0,0 +1,34 @@
+import statusCodes, { OK } from '../constants/status-codes.js';
+
+/**
+ * Checks if a page has a required radio answer before sending an event to the grant type machine
+ * @param {object} grantTypeMachineService - The grant type machine service
+ * @param {object} request - The request object
+ * @param {object} h - The response toolkit
+ * @returns {object} The response object
+ */
+export const checkPageForError = (grantTypeMachineService, request, h) => {
+  const { currentPageId, answer } = request.payload;
+
+  if (!answer && request.payload.questionType === 'radio') {
+    grantTypeMachineService.send({
+      type: 'UPDATE_STATE',
```

Comment:
Is this a new event type in the machine states as I do not see example grant machine changes?

Reply:
I think there must be a change, as the changes populate through so that an error message is displayed. However, your knowledge of the state machine is better, so if you think there's something to be added, let me know. The behaviour works.

Reply:
Is this supposed to introduce another event type and a state? Do we need an event being sent to the machine? I think we are not transitioning to a new state so we just need updating machine context with sth like:

1. Update context
```
context: {
  previousPageId: null,
  currentPageId: 'start',
  userAnswers: {},
  completedPageIds: [],
  validationErrors: {} // Track errors for the current page
}
```
2. Guards
```
guards: {
  isValidInput: (context, event) => {
    const { currentPageId, userAnswers } = context;
    const requiredAnswers = ['country']; // Define required fields per page

    // Check if all required fields have values
    return requiredAnswers.every((field) => userAnswers[currentPageId]?.[field]);
  }
}
```
3. Actions

```
actions: {
  setValidationErrors: assign({
    validationErrors: (context, event) => {
      const { currentPageId, userAnswers } = context;

      const requiredAnswers = ['country']; // Define required fields per page
      const errors = requiredAnswers
        .filter((field) => !userAnswers[currentPageId]?.[field])
        .reduce((acc, field) => {
          acc[field] = 'This field is required';
          return acc;
        }, {});

      return { [currentPageId]: errors };
    }
  }),

  clearValidationErrors: assign({
    validationErrors: (_context, _event) => {
      return {}; // Clear errors
    }
  })
}
```
4. Modify transitions

```
states: {
  country: {
    on: {
      NEXT: [
        {
          cond: 'isValidInput',
          target: 'consent',
          actions: ['trackPageCompletion', 'updateCurrentPageId', 'updateAnswers', 'clearValidationErrors']
        },
        {
          target: 'country',
          actions: ['setValidationErrors']
        }
      ],
      BACK: {
        target: 'start',
        actions: ['updateCurrentPageId']
      }
    },
    meta: {
      templateId: 'page',
      currentPageId: 'country',
      previousPageId: 'start',
      nextPageId: 'consent',
      title: 'Is the planned project in England?',
      hint: { text: 'The site where the work will happen' },
      questionType: 'radio',
      answers: [
        { key: 'country-A1', value: 'Yes' },
        { key: 'country-A2', value: 'No' }
      ]
    }
  }
}
```
WDYT?

Reply:
Thinking about it again isn't it better for client side validation with server side fallback?

Reply:
This has been pushed to state machine now.